purchase_sale_stock_inter_company/tests/test_inter_company_purchase_sale_stock.py

```diff
@@ -0,0 +1,440 @@
+                },
+            ),
+            (
+                0,
+                0,
+                {
+                    "location_id": so_move.location_id.id,
+                    "location_dest_id": so_move.location_dest_id.id,
+                    "product_id": self.stockable_product_serial.id,
+                    "quantity": 1,
+                    "lot_id": self.serial_2.id,
+                    "picking_id": so_picking_id.id,
+                },
+            ),
+            (
+                0,
+                0,
+                {
+                    "location_id": so_move.location_id.id,
+                    "location_dest_id": so_move.location_dest_id.id,
+                    "product_id": self.stockable_product_serial.id,
+                    "quantity": 1,
+                    "lot_id": self.serial_3.id,
+                    "picking_id": so_picking_id.id,
+                },
+            ),
+        ]
+        so_picking_id.sudo().button_validate()
+        so_lots = so_move.mapped("move_line_ids.lot_id")
+        po_lots = po_picking_id.mapped("move_ids.move_line_ids.lot_id")
+        self.assertEqual(
+            len(so_lots),
+            len(po_lots),
+            msg="There aren't the same number of lots on both moves",
+        )
+        self.assertNotEqual(
+            so_lots, po_lots, msg="The lots of the moves should be different objects"
+        )
+        self.assertEqual(
+            so_lots.mapped("name"),
+            po_lots.mapped("name"),
+            msg="The lots should have the same name in both moves",
+        )
+        self.assertIn(
+            serial_3_company_a,
+            po_lots,
+            msg="Serial 333 already existed, a new one shouldn't have been created",
+        )
+
+    def test_sync_picking_lot_and_qty_with_move_diff_2(self):
+        """
+        Test that the lot is synchronized on the moves
+        by searching or creating a new lot in the company of destination
+        """
+        self.purchase_company_a = (
+            self._create_purchase_order_for_stockable_product_serial(
+                self.partner_company_b, quantity=3
+            )
+        )
+        purchase = self.purchase_company_a
+        sale = self._approve_po()
+
+        # validate the SO picking
+        po_picking_id = purchase.picking_ids
+        so_picking_id = sale.picking_ids
+        so_move = so_picking_id.move_ids
+        so_move.move_line_ids = [
+            (
+                0,
+                0,
+                {
+                    "location_id": so_move.location_id.id,
+                    "location_dest_id": so_move.location_dest_id.id,
+                    "product_id": self.stockable_product_serial.id,
+                    "quantity": 1,
+                    "lot_id": self.serial_1.id,
+                    "picking_id": so_picking_id.id,
+                },
+            ),
+        ]
+        so_picking_id.sudo().button_validate()
```

Comment:
Test is failing here, because we try to deliver only 1 of the 3 requested products and button_validate is opening the dialog to create a backorder...

```suggestion
        res_dict = so_picking_id.sudo().button_validate()
        if isinstance(res_dict, dict) and "context" in res_dict:
            self.env["stock.backorder.confirmation"].with_context(
                **res_dict["context"]
            ).process()
```
I think using the same code as in previous tests with backorder creation would make this test pass.